baseplate/server/__init__.py

```diff
@@ -190,6 +206,54 @@ def configure_logging(config: Configuration, debug: bool) -> None:
         logging.config.fileConfig(config.filename)
 
 
+class BaseplateBatchSpanProcessor(BatchSpanProcessor):
+    def __init__(
+        self, otlp_exporter: OTLPSpanExporter, attributes: Optional[Dict[str, Any]] = None
+    ) -> None:
+        logger.info(
+            "Initializing %s with global attributes=%s.", self.__class__.__name__, attributes
+        )
+        super().__init__(otlp_exporter)
+        self.baseplate_global_attributes = attributes
+
+    def on_start(self, span: Span, parent_context: Optional[Context] = None) -> None:
+        logger.warning("Starting new span. [span=%s, parent_context=%s]", span, parent_context)
+        if self.baseplate_global_attributes:
+            span.set_attributes(self.baseplate_global_attributes)
+            logger.warning(
+                "Added global attributes to new span. [span=%s, baseplate_global_attributes=%s]",
+                span,
+                self.baseplate_global_attributes,
+            )
+        super().on_start(span, parent_context)
+
+
+def configure_tracing() -> None:
+    logger.info("Entering configure tracing function")
+    sample_rps = 10
+    try:
+        sample_rps = DefaultFromEnv(Integer, "BASEPLATE_TRACE_SAMPLER_RPS", 10)
+    except ValueError:
+        logger.warning(
+            "Invalid BASEPLATE_TRACE_SAMPLER_RPS, falling back to the default of 10 RPS."
+        )
```

Comment:
@chriskuehl I'm trying to use `DefaultFromEnv` as suggested but I think I'm misunderstanding how this is supposed to work. Does this also require implementing these options in the ini based config to get it to work?